src/bootctl/bootctl-random-seed.c

```diff
@@ -39,14 +39,12 @@ static int random_seed_verify_permissions(int fd, mode_t expected_type) {
         if ((st.st_mode & 0007) == 0) /* All world bits are off? Then all is good */
                 return 0;
 
-        if (S_ISREG(expected_type))
-                log_warning("%s Random seed file '%s' is world accessible, which is a security hole! %s",
-                            glyph(GLYPH_WARNING_SIGN), full_path, glyph(GLYPH_WARNING_SIGN));
-        else {
-                assert(S_ISDIR(expected_type));
-                log_warning("%s Mount point '%s' which backs the random seed file is world accessible, which is a security hole! %s",
-                            glyph(GLYPH_WARNING_SIGN), full_path, glyph(GLYPH_WARNING_SIGN));
-        }
+        log_error(S_ISREG(expected_type) ?
+                  "%s%sRandom seed file '%s' is world accessible, which is a security hole!%s%s" :
+                  "%s Mount point '%s' which backs the random seed file is world accessible, which is a security hole! %s",
```

Comment:
The second statement has fewer `%s`.

I do not like the format string is conditionalized. In such case, all format strings are checked by compilers? It seems not.

Reply:
You're right, this didn't make much sense. I changed it back the original form.